production/tanka/grafana-agent/v1/lib/logs.libsonnet

```diff
@@ -70,10 +70,10 @@ local container = k.core.v1.container;
       // For reading docker containers. /var/log is used for the positions file
       // and shouldn't be set to readonly.
       k.util.hostVolumeMount('varlog', '/var/log', '/var/log') +
-      k.util.hostVolumeMount('varlibdockercontainers', '/var/lib/docker/containers', '/var/lib/docker/containers', readOnly=true) +
+      k.util.hostVolumeMount('varlibdockercontainers', '/var/lib/docker/containers', '/var/lib/docker/containers', readOnly=true),
 
       // For reading journald
-      k.util.hostVolumeMount('etcmachineid', '/etc/machine-id', '/etc/machine-id', readOnly=true),
+      //k.util.hostVolumeMount('etcmachineid', '/etc/machine-id', '/etc/machine-id', readOnly=true),
```

Comment:
I'm not sure about commenting this out, because people still might want it. Can we make it opt-in instead? i.e., logsJournaldMixin or something?

Reply:
+1 for making this a parameter, and off by default.

Reply:
This also might be a good opportunity to pull the bandaid and migrate everything to the v2 jsonnet, removing the versions. This is already optionally configurable in v2.

Reply:
yeah, ok. agreed on pulling the bandaid and migrating to `v2` jsonnet. 

@rgeyer this will delay delivering pod logs a bit (but not by much). i think it's important as we still want to maintain this non-operator deployment path going forward.

another advantage is we can finally provide a statefulset as the default (and i can quickly write a config to turn on eventrouter that we can provide in the onboarding plugin)

closing this PR

@rfratto i couldn't find how to configure the `machine-id` vol in the v2 jsonnet (https://github.com/grafana/agent/blob/main/production/tanka/grafana-agent/v2/internal/helpers/logs.libsonnet#L13) - could you point me in the right direction?